services/src/main/scala/cromwell/services/metadata/impl/pubsub/PubSubMetadataServiceActor.scala

```diff
@@ -0,0 +1,133 @@
+  *  - Graceful shutdown could also be useful. The pubsub calls will autoretry in the face of transient errors which
+  *     means there might be stuff hanging around on a shutdown request.
+  *  - Currently service actors which fail to instantiate properly don't bring down Cromwell. It's easier to screw
+  *     it up here, we might want to revisit that strategy
+  *  - Revisit the Alpakka connector instead of the workbench-lib version. There was a bug they fixed but hadn't released
+  *     it yet. Also I'm pretty sure that same bug was repeated in a few other places.
+  */
+class PubSubMetadataServiceActor(serviceConfig: Config, globalConfig: Config) extends Actor with ActorLogging {
+  implicit val ec = context.dispatcher
+
+  // The auth *must* be a service account auth but it might not be named service-account.
+  val googleAuthName = serviceConfig.as[Option[String]]("auth").getOrElse("service-account")
+
+  val googleProject = serviceConfig.as[String]("project")
+  val pubSubTopicName = serviceConfig.as[Option[String]]("topic").getOrElse("cromwell-metadata")
+  val pubSubSubscriptionName = serviceConfig.as[Option[String]]("subscription")
+  val pubSubAppName = serviceConfig.as[Option[String]]("appName").getOrElse("cromwell")
+
+  val pubSubConnection = createPubSubConnection()
+  createTopicAndSubscription().failed foreach { e => throw e }
+
+  override def receive = {
+    case action: PutMetadataAction =>
+      publishMessages(action.events).failed foreach { e =>
+        log.error(e, "Failed to post metadata: " + action.events)
+      }
+    case action: PutMetadataActionAndRespond =>
+      publishMessages(action.events) onComplete {
+        case Success(_) => action.replyTo ! MetadataWriteSuccess(action.events)
+        case Failure(e) => action.replyTo ! MetadataWriteFailure(e, action.events)
+      }
+  }
+
+  protected[this] def createPubSubConnection(): GooglePubSubDAO = {
+    implicit val as = context.system
+
+    val googleConfig = GoogleConfiguration(globalConfig)
+
+    // This class requires a service account auth due to the library used
+    val googleAuth = googleConfig.auth(googleAuthName) match {
+      case Valid(a: ServiceAccountMode) => a
+      case Valid(doh) => throw new IllegalArgumentException(s"Unable to configure PubSubMetadataServiceActor: ${doh.name} was not a service account auth")
+      case Invalid(e) => throw new IllegalArgumentException("Unable to configure PubSubMetadataServiceActor: " + e.toList.mkString(", "))
+    }
+
+    val pemFile = googleAuth.fileFormat match {
+      case p: PemFileFormat => p
+      case _ => throw new IllegalArgumentException("Unable to configure PubSubMetadataServiceActor: the service account must supply a PEM file")
+    }
+
+    // The metric name is pretty useless. It's an artifact of the workbench libraries which we're not using
+    new HttpGooglePubSubDAO(pemFile.accountId, pemFile.file, pubSubAppName, googleProject, "cromwell.pubsub")
```

Comment:
_PEM files begone!_ Can this be switched to a JSON instead?

Reply:
Yeah. If it helps, me doing this project and then whining to them is what led to them getting into the 21st century